test/complex.spec.ts

```diff
@@ -0,0 +1,39 @@
+import { describe, it, expect, beforeEach } from 'jest-without-globals'
+import { getSnapshot } from 'mobx-state-tree'
+
+import { persist } from '../src/index'
+
+import { getItem } from './helpers'
+import { ComplexUserStoreF, persistedComplexUserDataF } from './fixtures'
+
+describe('persist complex types', () => {
+  beforeEach(() => window.localStorage.clear())
+
+  it('should persist an array', async () => {
+    const user = ComplexUserStoreF.create()
+    await persist('user', user)
+
+    user.addDog('Shadow') // fire action to trigger onSnapshot
+    expect(user.dogs).toStrictEqual(['Shadow'])
+    expect(getItem('user')).toStrictEqual(getSnapshot(user))
+  })
+
+  it('should persist a whitelisted array', async () => {
+    const user = ComplexUserStoreF.create()
+    await persist('user', user, {
+      whitelist: ['name', 'dogs']
```

Comment:
this is basically equivalent to the first test right now, though that might change as more is added to this store